src/main/java/echo/command/DeleteCommand.java

```diff
@@ -0,0 +1,55 @@
+package echo.command;
+
+import java.io.IOException;
+
+import echo.storage.Storage;
+import echo.task.TaskList;
+import echo.ui.Ui;
+import echo.utils.EchoException;
+
+/**
+ * This class inherits from the Command class and encapsulates the delete command.
+ */
+public class DeleteCommand extends Command {
+
+    /** String that represents the delete command. */
+    public static final String COMMAND = "delete";
+
+    /** Task index to delete. */
+    private final int TASK_INDEX;
+
+    /**
+     * Constructor for DeleteCommand.
+     *
+     * @param taskIndex Task index to delete.
+     */
+    public DeleteCommand(int taskIndex) {
+        this.TASK_INDEX = taskIndex;
+    }
+
+    /**
+     * Executes command.
+     *
+     * @param tasks TaskList containing list of tasks.
+     * @param ui Ui that deals with user interactions.
+     * @param storage Storage deals with loading tasks from the file and saving tasks in the file.
+     *
+     * @return String message representing command successful execution.
+     *
+     * @throws EchoException If input is invalid.
+     */
+    @Override
```

Comment:
If the method has an @/Override tag (ie the method is overriding a method from the parent class), I believe you can leave out the documentation :)